src/cluster/multi_step_cluster.cpp

```diff
@@ -92,43 +101,59 @@ void MultiStep::find_connected_components(vector<uint32_t>& sindex, unordered_ma
 
 
 	for (size_t i = 0; i < sindex.size(); i++) {
-		++comp[sindex[i]];
+		ne[sindex[i]] = {++ne[sindex[i]].nodes, ne[sindex[i]].edges += nedges[i]};
```

Comment:
Die beiden Anweisungen
`++ne[sindex[i]].nodes`
`ne[sindex[i]].edges += nedges[i]`
ändern bereits das Element, das in der map steht, d.h. du solltest dieses danach nicht nochmal überschreiben.